packages/shader-shaderlab/src/shaders/shadingPBR/BTDF.glsl

```diff
@@ -7,16 +7,30 @@
     sampler2D camera_OpaqueTexture;
     vec3 evaluateRefraction(SurfaceData surfaceData, BRDFData brdfData) {
         RefractionModelResult ray;
-        #ifdef REFRACTION_MODE
-            refractionModelBox(-surfaceData.viewDir, surfaceData.position, surfaceData.normal, surfaceData.IOR, 0.005, ray);
+        // '0' is refraction sphere, '1' is refraction plane, '2' is refraction thin
+        #if REFRACTION_MODE == 0
+            refractionModelSphere(-surfaceData.viewDir, surfaceData.position, surfaceData.normal, surfaceData.IOR, surfaceData.thickness, ray);
+        #elif REFRACTION_MODE == 1
+            refractionModelBox(-surfaceData.viewDir, surfaceData.position, surfaceData.normal, surfaceData.IOR, surfaceData.thickness, ray);
+        #elif REFRACTION_MODE == 2
+            refractionModelBox(-surfaceData.viewDir, surfaceData.position, surfaceData.normal, surfaceData.IOR, surfaceData.thickness, ray);
         #endif
-        //TODO: support cubemap refraction.
+
         vec3 refractedRayExit = ray.positionExit;
 
         // We calculate the screen space position of the refracted point
         vec4 samplingPositionNDC = camera_ProjMat * camera_ViewMat * vec4( refractedRayExit, 1.0 );
         vec2 refractionCoords = (samplingPositionNDC.xy / samplingPositionNDC.w) * 0.5 + 0.5;
 
+        // Absorption coefficient from Disney: http://blog.selfshadow.com/publications/s2015-shading-course/burley/s2015_pbs_disney_bsdf_notes.pdf
+       #ifdef MATERIAL_HAS_ABSORPTION
```

Comment:
We need to discuss: If need `MATERIAL_HAS_ABSORPTION` and `MATERIAL_HAS_THICKNESS` to opt